esphome/components/deep_sleep/deep_sleep_esp32.cpp

```diff
@@ -83,7 +83,11 @@ void DeepSleepComponent::deep_sleep_() {
     }
     gpio_sleep_set_direction(gpio_pin, GPIO_MODE_INPUT);
     gpio_hold_en(gpio_pin);
+#if !defined(USE_ESP32_VARIANT_ESP32C6)
```

Comment:
Wouldn't it be better to check for the config flag?
```suggestion
#if !SOC_GPIO_SUPPORT_HOLD_SINGLE_IO_IN_DSLP
```